etcdmain/config.go

```diff
@@ -317,6 +356,141 @@ func (cfg *config) Parse(arguments []string) error {
 		}
 	}
 
+	return nil
+}
+
+func (cfg *config) configFromFile() error {
+	// Read in the contents of the config file.
+	b, err := ioutil.ReadFile(cfg.configFile)
+	if err != nil {
+		return err
+	}
+
+	// Unmarshal the YAML.
+	err = yaml.Unmarshal(b, cfg)
+	if err != nil {
+		return err
+	}
+
+	// Make sure the defaults are set for the missing fields.
+	cfg.setDefaultsIfNeeded()
+
+	// Validate whether conflicting parameters are specified.
+	nSet := 0
```

Comment:
cmd and confFile shares some validation. probably we can move the shared validation part to `parse` function after we read out configuration?